src/lib.rs

```diff
@@ -664,9 +766,73 @@ impl Octocrab {
         self.client.request(method, url)
     }
 
-    /// Execute the given `request` octocrab's Client.
-    pub async fn execute(&self, request: reqwest::RequestBuilder) -> Result<reqwest::Response> {
-        request.send().await.context(error::Http)
+    /// Requests a fresh installation auth token and caches it. Returns the token.
+    async fn request_installation_auth_token(&self) -> Result<String> {
+        let (app, installation, token) = if let AuthState::Installation { ref app, installation, ref token } = self.auth_state {
+            (app, installation, token)
+        } else {
+            panic!("Installation not configured");
+        };
+        let mut retries = 0;
+        loop {
+            let result = self.client.post(self.absolute_url(format!("/app/installations/{}/access_tokens", installation))?)
```

Comment:
Shouldn't there be a `access_tokens` method here in the octocrab API that you can use instead of calling it directly?

Reply:
There could be, but as far as I can tell, this is the only reasonable way to use access tokens. You want them to auto-refresh, and you're going to use them to authenticate other Github requests, so why would you use a stand-alone `access_tokens` API call?

Reply:
Well someone wants to implement their own authentication for whatever reason, having this would make that slightly easier.